operator/main.go

```diff
@@ -148,6 +161,74 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	operatorID, err := os.Hostname()
+	if err != nil {
+		log.WithError(err).Fatal("failed to get hostname")
+	}
+	operatorID = rand.RandomStringWithPrefix(operatorID+"-", 10)
+
+	ns := option.Config.K8sNamespace
+	// If due to any reason the NAMESPACE is not set we assume it to be
+	// in default namespace.
+	if ns == "" {
+		ns = metav1.NamespaceDefault
+	}
+
+	leaseLock := &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      leaderElectionLeaseLockName,
+			Namespace: ns,
+		},
+		Client: k8s.Client().CoordinationV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			// Identity name of the holder
+			Identity: operatorID,
+		},
+	}
+
+	// Start the leader election for running cilium-operators
+	leaderelection.RunOrDie(leaderElectionCtx, leaderelection.LeaderElectionConfig{
+		Name: leaderElectionLeaseLockName,
+
+		Lock:            leaseLock,
+		ReleaseOnCancel: true,
+
+		LeaseDuration: operatorOption.Config.LeaderElectionLeaseDuration,
+		RenewDeadline: operatorOption.Config.LeaderElectionRenewDeadline,
+		RetryPeriod:   operatorOption.Config.LeaderElectionRetryPeriod,
+
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: onOperatorStartLeading,
+			OnStoppedLeading: func() {
+				log.WithField("operator_id", operatorID).Info("leader election lost")
```

Comment:
Nit: do you mind capitalizing the first word of each log msg? This helps keep the log formatting consistent across the codebase. (While I don't think every log msg in the codebase now starts with a capital, we should converge on one style for the sake of readability when reading logs. :) )

Reply:
Yeah, I think it makes sense to have this consistency. I will update the PR and will make sure this doesn't happen next time.